src/sema/expression.rs

```diff
@@ -5995,21 +5995,30 @@ fn method_call_pos_args(
                 ResolveTo::Type(&Type::Value),
             )?;
 
+            let address = cast(
+                &var_expr.loc(),
+                var_expr,
```

Comment:
Is there any case in which `loc` is not the same `loc` saved in an expression for the `cast` function? It seems redundant to have these two arguments as the former is an attribute of the latter.

Reply:
That is a good point, I had not considered that. 

I think you are totally right and this is remnant of the way the code used to be structured. 

Let me see if I remove the argument.

Reply:
I started this work and now I found a case where we use a different location for the first argument and the expression. You can see that here.

![image](https://user-images.githubusercontent.com/816900/155734480-2d9dca96-31f2-42f5-8486-0861d3a94571.png)